Libplanet/Blockchain/BlockChain.cs

```diff
@@ -727,6 +750,21 @@ bool renderActions
                 );
             }
 
+            if (block.Index == 0 && !block.Hash.Equals(_genesisHash))
+            {
+                string msg =
+                    $"The genesis block that the given {nameof(IStore)} contains does not match " +
+                    "to the genesis block that the network expects.  You might pass the wrong " +
+                    "store which is incompatible with this chain.  Or your network might " +
+                    "restarted the chain with a new genesis block so that it is incompatible " +
+                    "with your existing chain in the local store.";
```

Comment:
This message quite seems a dup of the exception message thrown by the constructor.